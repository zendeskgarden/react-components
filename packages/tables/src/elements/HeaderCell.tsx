/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { ThHTMLAttributes } from 'react';
import PropTypes from 'prop-types';

import { StyledHeaderCell, IStyledCellProps } from '../styled';
import { useTableContext } from '../utils/useTableContext';

/**
 * @extends ThHTMLAttributes<HTMLTableHeaderCellElement>
 */
export const HeaderCell = React.forwardRef<
  HTMLTableHeaderCellElement,
  Omit<IStyledCellProps, 'size'> & ThHTMLAttributes<HTMLTableHeaderCellElement>
>((props, ref) => {
  const { size } = useTableContext();

  return <StyledHeaderCell ref={ref} size={size} {...props} />;
});

HeaderCell.displayName = 'HeaderCell';

HeaderCell.propTypes = {
  isMinimum: PropTypes.bool,
  isTruncated: PropTypes.bool,
  hasOverflow: PropTypes.bool,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};
