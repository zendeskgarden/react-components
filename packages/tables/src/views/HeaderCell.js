/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import Cell from './Cell';

const COMPONENT_ID = 'tables.header_cell';

/** Accepts all `<div>` props */
const HeaderCell = styled(Cell).attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  role: 'columnheader'
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

HeaderCell.propTypes = {
  /** Applies minimum size styling */
  minimum: PropTypes.bool,
  /** Applies truncated text styling */
  truncate: PropTypes.bool,
  /** Applies overflow styling */
  menu: PropTypes.bool,
  /** The width of the cell */
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

/** @component */
export default HeaderCell;
