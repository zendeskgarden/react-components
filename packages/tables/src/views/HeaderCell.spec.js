/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import { shallow } from 'enzyme';

import HeaderCell from './HeaderCell';

describe('HeaderCell', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<HeaderCell />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders minimum styling if provided', () => {
    const wrapper = shallow(<HeaderCell minimum />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders truncation styling if provided', () => {
    const wrapper = shallow(<HeaderCell truncate />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders menu styling if provided', () => {
    const wrapper = shallow(<HeaderCell menu />);

    expect(wrapper).toMatchSnapshot();
  });
});
