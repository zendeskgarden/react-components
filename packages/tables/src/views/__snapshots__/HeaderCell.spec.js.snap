// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCell renders default styling 1`] = `
<Cell
  className=""
  data-garden-id="tables.header_cell"
  data-garden-version="version"
  role="columnheader"
/>
`;

exports[`HeaderCell renders menu styling if provided 1`] = `
<Cell
  className=""
  data-garden-id="tables.header_cell"
  data-garden-version="version"
  menu={true}
  role="columnheader"
/>
`;

exports[`HeaderCell renders minimum styling if provided 1`] = `
<Cell
  className=""
  data-garden-id="tables.header_cell"
  data-garden-version="version"
  minimum={true}
  role="columnheader"
/>
`;

exports[`HeaderCell renders truncation styling if provided 1`] = `
<Cell
  className=""
  data-garden-id="tables.header_cell"
  data-garden-version="version"
  role="columnheader"
  truncate={true}
/>
`;
